server/src/main/java/org/elasticsearch/common/lucene/Lucene.java

```diff
@@ -113,7 +114,8 @@ private Lucene() {}
      * Reads the segments infos, failing if it fails to load
      */
     public static SegmentInfos readSegmentInfos(Directory directory) throws IOException {
-        return SegmentInfos.readLatestCommit(directory);
+        // Temporary workaround
+        return SegmentInfos.readLatestCommit(directory, IndexVersions.MINIMUM_READONLY_COMPATIBLE.luceneVersion().major);
```

Comment:
why is this temporary? I thought it's good to go.

Reply:
I mostly wanted your validation on this, now I have it :+1: